src/Entity/Components/TargetTowerComponent.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <flecs.h>
+
+namespace game
+{
+
+//I split this in two components, cause player target is always same and can be inherited, and tower will be changed all the time
+//But on the other hand, it's just a pointer, no harm in dragging it around? But at this point
+//I follow: make the minimal possible division
```

Comment:
Let’s not go too specific either, it’s good to have components for each « action », but if they’re really similar we should try to refactor.

For now, `TargetPlayerComponent` doesn’t seem to be used so we can remove it, and rename this one to `TargetComponent`. 

Later we can make use of `_second` functions to have several times the same component type on an entity, so like we could do:
```cpp
entity.emplace_second<TargetComponent>(TowerTag, towerId);
```

cf: https://github.com/HypixelStudios/HytaleEngine/blob/dev/Documentation/ThirdParty/flecs/flecs_guidelines.md#g10-prefer-the-use-of-_first-and-_second-functions-over-creating-child-entities-to-store-several-components-on-a-same-entity